xc/xenops_server_xen.ml

```diff
@@ -1581,6 +1581,27 @@ module VM = struct
 									(fun fd ->
 										Domain.restore task ~xc ~xs ~store_domid ~console_domid ~no_incr_generationid (* XXX progress_callback *) ~timeoffset ~extras build_info (choose_xenguest vm.Vm.platformdata) domid fd
 									);
+								if inject_igmp_query then
+									try
```

Comment:
Absolutely. I meant that it should be in its own try/with, and not together with `Domain.restore`.